msmbuilder/featurizer/featurizer.py

```diff
@@ -307,6 +307,73 @@ def partial_transform(self, traj):
         return d ** self.exponent
 
 
+class FunctionFeaturizer(Featurizer):
+    """Featurizer based on arbitrary functions.
+
+    This featurizer transforms a dataset containing MD trajectories into
+    a vector dataset by representing each frame in each of the MD trajectories
+    by a vector the output of the function.
+
+    Parameters
+    ----------
+    function : Instantiation of the function
+    *args : arguments to pass to the function MINUS the trajectory keyword
+    **kwargs: key word arguments to pass to the function MINUS the trajectory
+    keyword
+
+    UseCase:
```

Comment:
Should i move it to the notes? I feel having an example here is going to be useful.